prover_rust/src/geth_client/types.rs

```diff
@@ -0,0 +1,41 @@
+use eth_types::{H256, U64};
+use serde::{Deserialize, Serialize};
+
+use crate::types::CommonHash;
+use prover::BlockTrace as ProverBlockTrace;
+
+
+/// l2 block full trace
+#[derive(Deserialize, Serialize, Default, Debug, Clone)]
+pub struct BlockTrace {
+    #[serde(flatten)]
+    pub block_trace: ProverBlockTrace,
+
+    pub version: String,
+
+    pub withdraw_trie_root: Option<CommonHash>,
+
+    #[serde(rename = "mptwitness", default)]
+    pub mpt_witness: Vec<u8>,
+}
+
+pub fn get_block_number(block_trace: &ProverBlockTrace) -> Option<u64> {
+    block_trace.header.number.map(|n| n.as_u64())
+}
+
+pub type TxHash = H256;
+
+/// this struct is tracked to https://github.com/scroll-tech/go-ethereum/blob/0f0cd99f7a2e/core/types/block.go#Header
```

Comment:
this will not change very often (especially for the classic fields, they are rarely changed).

the struct is changed for compatibility issues (e.g., need to decode new fields using the SDK in l2geth repo).

but yeah, if sync upstream in done, the header field would be changed again.

Reply:
actually new fields in the l2geth SDK is not used here, the only used field is wrapped by a get_number function.
this implementation is for making things clear and for future requirements maybe